### ISSUE TITLE
List regex flavor options in README

### ISSUE THREAD

**Issue by tecfu** [ggreer/the_silver_searcher#697]
What families of regular expressions is Ag compatible with? i.e. POSIX, POSIX Extended, Vim, Perl...

This isn't immediately clear...there's no definitive answer online, the man page says nothing on the subject. If stated in the README and it would save new users lots of time and confusion and give instant clarity to new users.

The only answer I've found is on stack overflow and comes from a non-definitive source. That answer states says Ag uses PCRE, 

http://stackoverflow.com/questions/22190334/ag-silver-searcher-what-are-the-rules-for-patterns

but that's clearly not always the case, as I can use POSIX character classes from my terminal like so

```
ag [[:digit:]] src/*.js
```

**Comment by decaff** [comment]
From looking at the code...

Ag uses the pcre(3) library with the PCRE_MULTILINE option enabled.  Regex features are listed here:       

````
man 3 pcrepattern
````

**Comment by decaff** [comment]
A man page update is available in [this commit](https://github.com/decaff/the_silver_searcher/commit/f45828812cfe82c3e2dce9fd8c0297fab54de3a6) .

**Comment by tecfu** [comment]
Great work! I hope to see that get merged.

**Comment by tgroshon** [comment]
Did a PR get submitted or merged for this?

**Comment by decaff** [comment]
No.  Geoff is pretty busy.  I'm letting my commits accumulate.  I'm sure it will be sorted when he has some free time.
